x/market/keeper/deal.go

```diff
@@ -125,3 +125,108 @@ func (k Keeper) SetDeal(ctx sdk.Context, deal types.Deal) {
 	bz := k.cdc.MustMarshalBinaryLengthPrefixed(&deal)
 	store.Set(dealKey, bz)
 }
+
+func (k Keeper) SellOwnData(ctx sdk.Context, seller sdk.AccAddress, cert types.DataValidationCertificate) (sdk.Coin, error) {
+	err := k.isDataCertDuplicate(ctx, cert)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+
+	findDeal, err := k.GetDeal(ctx, cert.UnsignedCert.GetDealId())
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+
+	if findDeal.GetStatus() == INACTIVE || findDeal.GetStatus() == COMPLETED {
+		return sdk.Coin{}, sdkerrors.Wrap(err, "deal is not activated")
+	}
```

Comment:
What I meant was to separate these two cases.
I thought it would be better to response fail message which informs the current state of deal, `INACTIVE` or `COMPLETED`
If you think no need to distinguish these two cases, the previous code you wrote `if findDeal.GetStatus() != ACTIVE` would be enough.

Reply:
Ah I understand it. I will change the errors message.